components/production/editor/ProductionEditor.vue

```diff
@@ -2,50 +2,134 @@
+              >Change</sta-button
+            >
+          </template>
+          <template v-else>
+            <form-label>
+              Slug
+              <t-input
+                :value="manualSlug"
+                @input="manualSlug = kebabCase($event)"
+              />
+            </form-label>
+            <sta-button
+              class="bg-sta-green hover:bg-sta-green-dark transition-colors"
+              @click="
+                () => {
+                  $emit('update:slug', manualSlug)
+                  changingSlug = false
+                }
+              "
+              >Done</sta-button
+            >
+          </template>
+          <br />
+          <error-helper :errors="errors" field-name="slug" />
+        </p>
+        <form-label :errors="errors" name="subtitle">
           Subtitle
-          <t-input v-model="editingProduction.subtitle" />
+          <t-input
+            :value="subtitle"
+            @input="$emit('update:subtitle', $event)"
+          />
         </form-label>
-        <form-label>
+        <form-label :errors="errors" name="description">
           Description <required-star />
           <template #control>
-            <rich-text-input v-model="editingProduction.description" />
+            <rich-text-input
+              :value="description"
+              @input="$emit('update:description', $event)"
+            />
           </template>
         </form-label>
-        <form-label>
-          Audience Warnings <required-star />
+        <form-label :errors="errors" name="warnings">
+          Audience Warnings
           <template #control>
             <div class="flex flex-wrap space-x-3">
               <div v-for="(warning, index) in availableWarnings" :key="index">
-                <input type="checkbox" /> {{ warning.warning }}
+                <input
+                  type="checkbox"
+                  :checked="
+                    warnings.find(
+                      (currentWarning) => currentWarning.id === warning.id
+                    )
+                  "
+                  @change="updateWarnings(warning, $event.target.checked)"
+                />
+                {{ warning.description }}
               </div>
             </div>
           </template>
         </form-label>
-        <form-label>
+        <form-label :errors="errors" name="facebookEvent">
           Facebook Event Link
-          <t-input v-model="editingProduction.facebookEvent" />
+          <t-input
+            :value="facebookEvent"
+            @input="$emit('update:facebookEvent', $event)"
+          />
         </form-label>
-        <form-label>
+        <form-label :errors="errors" name="ageRating">
           Age Rating
           <t-input
-            v-model="editingProduction.ageRating"
+            :value="ageRating"
             type="number"
             min="4"
             max="18"
+            @input="$emit('update:ageRating', $event)"
           />
         </form-label>
```

Comment:
a validator for checking the input is an integer, and it doesnt break when "18+" is entered - this i feel is likely may happen

Reply:
Do you want to try and do this one? Should be fairly straight forward